rca/projects/models.py

```diff
@@ -265,4 +273,164 @@ def get_context(self, request, *args, **kwargs):
+                    "filter_type": "school_or_centre",
+                }
+            )
+        filters.append(school_or_centre)
+        for i in ResearchCentrePage.objects.all():
+            school_or_centre["items"].append(
+                {
+                    "id": i.id,
+                    "title": i.title,
+                    "link": slugify(i.title),
+                    "filter_type": "school_or_centre",
+                }
+            )
+        filters.append(school_or_centre)
+
+        return filters
+
+    def _format_results(self, projects):
+        """ Prepares the queryset into a digestable list for the template """
+        projects_formatted = []
+        for page in projects:
+            year = None
+            if page.start_date:
+                year = page.start_date.strftime("%Y")
+                if page.end_date and page.end_date.strftime("%Y") != year:
+                    end_year = page.end_date.strftime("%Y")
+                    year = year + " - " + end_year
+
+            projects_formatted.append(
+                {
+                    "title": page.title,
+                    "image": page.hero_image,
+                    "link": page.url,
+                    "school": page.get_related_school(),
+                    "year": year,
+                }
+            )
+        return projects_formatted
+
+    def get_extra_query_params(self, request):
+        # Formats the request parameters in a list
+        extra_query_params = []
+        for i in request.GET.getlist("type"):
+            extra_query_params.append(urlencode({"type": i}))
+        for i in request.GET.getlist("subject"):
+            extra_query_params.append(urlencode({"subject": i}))
+        for i in request.GET.getlist("school_or_centre"):
+            extra_query_params.append(urlencode({"school_or_centre": i}))
+        return extra_query_params
+
+    def get_results(self, request, context):
+        projects = (
+            ProjectPage.objects.live()
+            .public()
+            .descendant_of(self, inclusive=True)
+            .select_related("hero_image")
+        )
+
+        # Request filters
+        research_types = request.GET.getlist("type")
+        subjects = request.GET.getlist("subject")
+        school_or_centre = request.GET.getlist("school_or_centre")
+
+        if research_types:
+            projects = projects.filter(
+                research_types__research_type_id__in=research_types
+            )
+
+        if subjects:
+            projects = projects.filter(subjects__subject_id__in=subjects)
+
+        if school_or_centre:
+            projects = projects.filter(
+                models.Q(related_school_pages__page_id__in=school_or_centre)
+                | models.Q(related_research_pages__page_id__in=school_or_centre)
+            )
+        return self._format_results(projects.distinct())
+
+    def get_context(self, request, *args, **kwargs):
+        page = request.GET.get("page", 1)
```

Comment:
This will be changed to a settings var, kept it like this for testing

Reply:
@kevinhowbrook could you explain what will become a setting here?

Reply:
https://github.com/torchbox/rca-wagtail-2019/blob/master/rca/settings/base.py#L672

Reply:
(the default count of items... which I think should be 25)

Reply:
Ahh, the confusion was caused by the line this comment is referring to, it should be the line: `paginator = Paginator(project_results, 24)`

I will fix this now so that we use `DEFAULT_PER_PAGE` which should be 24

Reply:
duh, yea my bad. Thanks